src/main/java/net/minestom/server/command/builder/ArgumentMapper.java

```diff
@@ -0,0 +1,26 @@
+package net.minestom.server.command.builder;
+
+import net.minestom.server.command.builder.exception.ArgumentSyntaxException;
+
+/**
+ * Represents a lambda that can turn an input into an output
+ * that also allows the throwing of ArgumentSyntaxException
+ *
+ * @param <I> The input expected from the Argument
+ * @param <O> The desired output type from this lambda.
+ */
+@FunctionalInterface
+public interface ArgumentMapper<I, O> {
```

Comment:
Why not just use a `Function`?

Reply:
Because it can throw an error -- originally I did use a function but exceptions cant be thrown in it without being caught inside the lambda.